ide/static/ide/js/resources.js

```diff
@@ -314,11 +324,28 @@ CloudPebble.Resources = (function() {
             var form = pane.find('form');
             form.submit(function(e) {
                 e.preventDefault();
-                process_resource_form(form, false, "/ide/project/" + PROJECT_ID + "/resource/" + resource.id + "/update", function(data) {
+                process_resource_form(form, false, resource.file_name, "/ide/project/" + PROJECT_ID + "/resource/" + resource.id + "/update", function(data) {
                     // Update any previews we have.
-                    $('.image-resource-preview img').each(function() {
-                        $(this).attr('src', $(this).attr('src').replace(/#e.*$/,'') + '#e' + (++preview_count));
+                    _.each(data.variants, function(variant) {
+                        var variant_div = $('.variant-'+variant);
+                        variant_div.find('img').attr('src', function(ind, old_src) {
+                            if (!old_src || /^(\s*(\?t=.*)|\s+)$/.test(old_src)) {
+                                // If we're uploading a new variant, we need to set its preview url from scratch
+                                return '/ide/project/' + PROJECT_ID + '/resource/' + resource.id + '/' + variant + '/get';
+                            }
+                            else {
+                                // If not, we have to refresh and bypass the cache the image using a query string
+                                return old_src.replace(/\?t=.*$/,'') + '?t=' + (++preview_count);
+                            }
+                        });
+                        if (resource.kind == 'png' || resource.kind == 'png-trans') {
```

Comment:
Is this the best place to make this check? If we will never show it, why are we bothering to try assembling things to put in it?